js/planning.js

```diff
@@ -273,99 +274,105 @@ var GLPIPlanning  = {
                     $('.planning-context-menu').remove();
 
                     // create new one
-                    var context = $('<ul class="planning-context-menu" data-event-id=""> \
-                  <li class="clone-event"><i class="far fa-clone"></i>'+__("Clone")+'</li> \
-                  <li class="delete-event"><i class="fas fa-trash"></i>'+__("Delete")+'</li> \
-               </ul>');
-
-                    // add it to body and place it correctly
-                    $('body').append(context);
-                    context.css({
-                        left: offset.left + element.outerWidth() / 4,
-                        top: offset.top
-                    });
+                    var actions = '';
+                    if (options.can_create) {
+                        actions += '<li class="clone-event"><i class="far fa-clone"></i>'+__("Clone")+'</li>';
+                    }
+                    if (options.can_delete) {
+                        actions += '<li class="delete-event"><i class="fas fa-trash"></i>'+__("Delete")+'</li>';
+                    }
+                    if (actions != '') {
```

Comment:
Using following code would limit whitespace changes, and may improve code readability.
```js
if (actions == '') {
    return;
}

// existing code
```